src/routes/SafeAppLandingPage/SafeAppLandingPage.tsx

```diff
@@ -0,0 +1,321 @@
+import { userAccountSelector } from 'src/logic/wallets/store/selectors'
+import { useStateHandler } from 'src/logic/hooks/useStateHandler'
+import { setChainId } from 'src/logic/config/utils'
+import ProviderDisconnected from 'src/components/AppLayout/Header/components/ProviderInfo/ProviderDisconnected'
+import Provider from 'src/components/AppLayout/Header/components/Provider'
+import ConnectDetails from 'src/components/AppLayout/Header/components/ProviderDetails/ConnectDetails'
+import NetworkLabel from 'src/components/NetworkLabel/NetworkLabel'
+import Img from 'src/components/layout/Img'
+import Link from 'src/components/layout/Link'
+
+const DEMO_SAFE_MAINNET = '0xfF501B324DC6d78dC9F983f140B9211c3EdB4dc7'
+
+const demoSafeAppsPath = generateSafeRoute(SAFE_ROUTES.APPS, {
+  shortName: 'eth',
+  safeAddress: DEMO_SAFE_MAINNET,
+})
+
+function SafeAppLandingPage(): ReactElement {
+  const { search } = useLocation()
+  const query = new URLSearchParams(search)
+  const safeAppChainId = query.get('chainId')
+  const safeAppUrl = query.get('appUrl')
+
+  // if no valid chainId or Safe App is present in query params we redirect to the Welcome page
+  useEffect(() => {
+    const isValidChainId = safeAppChainId && getChains().find((chain) => chain.chainId === safeAppChainId)
+    const redirectToWelcome = !safeAppUrl || !isValidChainId
+    if (redirectToWelcome) {
+      history.push(WELCOME_ROUTE)
+    }
+
+    // we set the valid Safe App chainId in the state
+    if (isValidChainId) {
+      setChainId(safeAppChainId)
+    }
+  }, [safeAppChainId, safeAppUrl])
+
+  const userAddress = useSelector(userAccountSelector)
+  const isWalletConnected = !!userAddress
+  const { clickAway, open, toggle } = useStateHandler()
+
+  // fetch Safe App details from the Config service
+  const { appList, isLoading: isConfigServiceLoading } = useAppList()
+  const safeAppDetailsFromConfigService = appList.find(({ url }) => safeAppUrl === url)
+
+  // fetch Safe App details from Manifest.json
+  const [safeAppDetailsFromManifest, setSafeAppDetailsFromManifest] = useState<SafeApp>()
+  const [isManifestLoading, setIsManifestLoading] = useState(false)
+  const [isManifestError, setIsManifestError] = useState(false)
+  useEffect(() => {
+    const fetchManifest = !isManifestLoading && !safeAppDetailsFromManifest && safeAppUrl && !isManifestError
+    if (fetchManifest) {
+      const fetchSafeAppDetailsFromManifest = async () => {
+        setIsManifestLoading(true)
+        try {
+          const safeAppDetailsFromManifest = await getAppInfoFromUrl(safeAppUrl)
+          setIsManifestLoading(false)
+          setSafeAppDetailsFromManifest(safeAppDetailsFromManifest)
+        } catch (e) {
+          setIsManifestLoading(false)
+          setIsManifestError(true)
+        }
+      }
+
+      fetchSafeAppDetailsFromManifest()
+    }
+  }, [isManifestLoading, safeAppDetailsFromManifest, safeAppUrl, isManifestError])
```

Comment:
You should check that the current effect `isCurrent` before setting the state here.

Reply:
Are you sure? I think that `safeAppUrl` never changes so I think that in this case is not necessary.

Reply:
We have a hook, `useAsyncValue`. It's worthwhile doing it with every fetch.

Reply:
ok, to be able to use the `useAsyncValue` here, I need to add a `isLoading` state.

Reply:
Alright, you need not use that hook then. I still think you should do an `isCurrent` check nonetheless.

Reply:
I mean I am using the `useAsyncValue`, but I updated it to add a `isLoading` state. [You can check it here](https://github.com/gnosis/safe-react/pull/3764/files#diff-9faeba421133c1bff7dc9137804984179612a9ce45ea9ba455c6d44cd6271265)

Reply:
Apologise. As per @usame-algan's [comment](https://github.com/gnosis/safe-react/pull/3764/files#r846312916), you don't need to add a new state for it.

Reply:
mm I think that can be issues if the promise is rejected with an `undefined` value, so I think that the best option here is to add this state to make sure that we have a 100% generic hook. [see more details here](https://github.com/gnosis/safe-react/pull/3764#discussion_r846326626)

Reply:
Very true. Nice one for catching us on that!